### ISSUE TITLE
Mechanism for failed init()

### ISSUE THREAD

**Issue by galou** [davetcoleman/ros_control_boilerplate#3]
The function init() has prototype`void ros_control_boilerplate::GenericHWInterface::init()`. There is thus no way to specify a failed hardware initialization other than throwing an exception. Is this the intended way to do it? Maybe a `bool ros_control_boilerplate::GenericHWInterface::init()` would make sense?

**Comment by davetcoleman** [comment]
Good point. For code this low level, though, is there any error handling you would do if you received a `false`? Could you not just add a loop inside `init()` that continued to attempt to reconnect to your hardware until you got the response you wanted?

**Comment by galou** [comment]
I don't need an error handling outside the init() function but from what you say, the standard way to do it is to loop inside the init() function, probably better with a ROS_WARNING, until the connection to the hardware is successful.
If I understand the philosophy, the only way one should go out of the init() function is after success in connecting to the hardware.

**Comment by davetcoleman** [comment]
I'm not certain I'm right or wrong, and there probably isn't a standard philosophy at this point. Are you saying you really want a `bool` return value?
